homeassistant/components/husqvarna_automower_ble/__init__.py

```diff
@@ -0,0 +1,67 @@
+"""The Husqvarna Autoconnect Bluetooth integration."""
+
+from __future__ import annotations
+
+import logging
+
+from automower_ble.mower import Mower
+from bleak_retry_connector import close_stale_connections_by_address, get_device
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, CONF_CLIENT_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import Coordinator
+
+LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [
+    Platform.LAWN_MOWER,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Husqvarna Autoconnect Bluetooth from a config entry."""
+    address = entry.data[CONF_ADDRESS]
+    channel_id = entry.data[CONF_CLIENT_ID]
+
+    mower = Mower(channel_id, address)
+
+    await close_stale_connections_by_address(address)
+
+    LOGGER.debug("connecting to %s with channel ID %s", address, str(channel_id))
+    device = bluetooth.async_ble_device_from_address(
+        hass, address, connectable=True
+    ) or await get_device(address)
+    if not await mower.connect(device):
+        return False
```

Comment:
I don't think we want to "fail" the integration here.  We have already "validated" the connectivity in the config flow.  At this point, it's just not in range and should continue to try.  Simply `raise  ConfigEntryNotReady()` to let HA retry automatically.